examples/meta/generator/targets/cpp.json

```diff
@@ -0,0 +1,61 @@
+{
+    "Program": "#include <shogun/base/init.h>\n#include <shogun/io/SGIO.h>\n#include <shogun/features/DenseFeatures.h>\n#include <shogun/labels/MulticlassLabels.h>\n#include <shogun/multiclass/KNN.h>\n#include <shogun/io/CSVFile.h>\n#include <shogun/distance/EuclideanDistance.h>\n\n${dependencies}\n\nusing namespace shogun;\n\nint main(int, char*[]) {\n    init_shogun_with_defaults();\n\n$program\n    exit_shogun();\n    return 0;\n}\n",
+    "Dependencies": {
+        "AllDependencies": "$enumDependencies\n\n",
+        "EnumDependencies": "$enums",
+        "DependencyListElementClass": "$element",
+        "DependencyListElementEnum": "import static org.shogun.$type.$value;",
+        "DependencyListSeparator": "\n"
+    },
+    "Statement": "    $statement;\n",
+    "Comment": "    //$comment\n",
+    "Init": {
+        "Construct": "C$type* $name = new C$type($arguments);\n    SG_REF($name)",
```

Comment:
I think we need to distribuish constructors for SGObject (with Cprefix) and other objects